src/tilesource.js

```diff
@@ -217,12 +226,49 @@ $.TileSource.prototype = /** @lends OpenSeadragon.TileSource.prototype */{
      * Subclasses should override this if tileSizes can be different at different levels
      *   such as in IIIFTileSource.  Code should use this function rather than reading
      *   from .tileSize directly.  tileSize may be deprecated in a future release.
+     * @deprecated
      * @function
      * @param {Number} level
      */
     getTileSize: function( level ) {
+        $.console.error(
+            "[TileSource.getTileSize] is deprecated." +
+            "Use TileSource.getTileWidth() and TileSource.getTileHeight() instead"
+        );
         return this.tileSize;
     },
+    
+    /**
+     * Return the tileWidth for a given level.
+     * Subclasses should override this if tileWidth can be different at different levels
+     *   such as in IIIFTileSource.  Code should use this function rather than reading
+     *   from .tileWidth directly.
+     * @function
+     * @param {Number} level
+     */
+    getTileWidth: function( level ) {
+        // If tileWidth was not set, fallback by setting it to tileSize
+        if( this.tileWidth === undefined){
```

Comment:
I guess we need to accommodate the fact that a TileSource subclass might have getTileSize() and not understand getTileWidth(), so maybe here we do 

```
if (!this._tileWidth) {
    return this.getTileSize(level);
}
```

Note that we need to call it every time, because the level might be different. 

Same for getTileHeight of course. 

Btw, even though we don't need to rewrite the IIIFTileSource because we're supporting this fallback, it's probably still a good idea to do so as a good example for future TileSource subclassers.